preview/index.js

```diff
@@ -0,0 +1,78 @@
+const commandLineArgs = require('command-line-args');
+const path = require('path');
+const matter = require('gray-matter');
+const express = require('express');
+const octokit = require('@octokit/rest')();
+const createPipieline = require('../src/site/stages/preview');
+
+const ENVIRONMENTS = require('../src/site/constants/environments');
+const HOSTNAMES = require('../src/site/constants/hostnames');
+
+const defaultBuildtype = ENVIRONMENTS.LOCALHOST;
+const defaultHost = HOSTNAMES[defaultBuildtype];
+const defaultContentDir = '../../../../../vagov-content/pages';
+
+const COMMAND_LINE_OPTIONS_DEFINITIONS = [
+  { name: 'buildtype', type: String, defaultValue: defaultBuildtype },
+  { name: 'buildpath', type: String, defaultValue: 'build/localhost' },
+  { name: 'host', type: String, defaultValue: defaultHost },
+  { name: 'port', type: Number, defaultValue: 3001 },
+  { name: 'entry', type: String, defaultValue: null },
+  { name: 'protocol', type: String, defaultValue: 'http' },
+  { name: 'destination', type: String, defaultValue: null },
+  { name: 'content-directory', type: String, defaultValue: defaultContentDir },
+  { name: 'unexpected', type: String, multile: true, defaultOption: true },
+];
+
+const options = commandLineArgs(COMMAND_LINE_OPTIONS_DEFINITIONS);
+
+if (options.unexpected && options.unexpected.length !== 0) {
+  throw new Error(`Unexpected arguments: '${options.unexpected}'`);
+}
+
+const app = express();
+
+app.use(express.static(path.join(__dirname, '..', options.buildpath)));
+
+app.use('/preview', (req, res) => {
+  const smith = createPipieline({
+    ...options,
+    port: process.env.PORT,
+  });
+
+  const contentId = req.query.contentId;
+
+  octokit.repos
+    .getContents({
+      owner: 'department-of-veterans-affairs',
+      repo: 'vagov-content',
+      path: `/pages/${contentId}`,
+    })
+    .then(result => Buffer.from(result.data.content, result.data.encoding))
+    .then(matter)
+    .then(parsedContent => {
+      const files = {
+        [contentId]: Object.assign(parsedContent.data, {
+          path: contentId,
+          contents: new Buffer(parsedContent.content),
+        }),
+      };
+
+      smith.run(files, (err, newFiles) => {
+        if (err) throw err;
+        res.set('Content-Type', 'text/html');
+        // This will actually need to convert a md path to an html path, probably
+        res.send(Object.entries(newFiles)[0][1].contents);
+      });
+    })
+    .catch(err => {
+      // eslint-disable-next-line no-console
+      console.log(err);
+      res.sendStatus(500);
+    });
+});
+
+app.listen(process.env.PORT || options.port, () => {
+  // eslint-disable-next-line no-console
+  console.log('Content preview server running');
+});
```

Comment:
Would it make sense for this to also live in `src/site/preview` or `script`?, or possibly `src/platform`? I'd like to keep the root of the repo pretty slim if possible.

Reply:
Yeah, I think I can put it back in script. It felt weird to put it in `src/`, since it's pretty separate from most of our source code.